tests/test_model_v2.py

```diff
@@ -596,6 +596,150 @@ def test_traits_datetime():
     assert str(e_info.value).startswith('Cannot decode datetime from value xyz')
 
 
+def test_traits_datetimeoffset():
+    """Test Edm.DateTimeOffset traits"""
+
+    typ = Types.from_name('Edm.DateTimeOffset')
+    assert repr(typ.traits) == 'EdmDateTimeOffsetTypTraits'
+
+    # 1. direction Python -> OData
+
+    testdate = datetime(2005, 1, 28, 18, 30, 44, 123456, tzinfo=timezone(timedelta(hours=3, minutes=40)))
+    assert typ.traits.to_literal(testdate) == "datetimeoffset'2005-01-28T18:30:44.123456+03:40'"
+
+    # without milliseconds part, negative offset
+    testdate = datetime(2005, 1, 28, 18, 30, 44, 0, tzinfo=timezone(-timedelta(minutes=100)))
+    assert typ.traits.to_literal(testdate) == "datetimeoffset'2005-01-28T18:30:44-01:40'"
+
+    # serialization of invalid value
+    with pytest.raises(PyODataModelError) as e_info:
+        typ.traits.to_literal('xyz')
+    assert str(e_info.value).startswith('Cannot convert value of type')
+
+    # 2. direction Literal -> python
+
+    # parsing full representation, with positive timezone offset
+    testdate = typ.traits.from_literal("datetimeoffset'1976-11-23T03:33:06.654321+12:11'")
+    assert testdate.year == 1976
+    assert testdate.month == 11
+    assert testdate.day == 23
+    assert testdate.hour == 3
+    assert testdate.minute == 33
+    assert testdate.second == 6
+    assert testdate.microsecond == 654321
+    assert testdate.tzinfo == timezone(timedelta(hours=12, minutes=11))
+
+    # parsing without milliseconds, negative offset, lowercase 'T'
+    testdate = typ.traits.from_literal("datetimeoffset'1976-11-23t03:33:06-01:00'")
+    assert testdate.year == 1976
+    assert testdate.second == 6
+    assert testdate.microsecond == 0
+    assert testdate.tzinfo == timezone(-timedelta(hours=1))
+
+    # UTC
+    testdate = typ.traits.from_literal("datetimeoffset'1976-11-23T03:33:06Z'")
+    assert testdate.year == 1976
+    assert testdate.second == 6
+    assert testdate.microsecond == 0
+    assert testdate.tzinfo == timezone.utc
+
+    # UTC lowercase
```

Comment:
where us the lowercase 'z' test case coming from? XML schema defines only 'Z' - see section 3.2.7.3 Timezones, cite:

_The lexical representation of a timezone is a string of the form: (('+' | '-') hh ':' mm) | 'Z',_  

On the other hand, for the UTC timezone, two other offset test cases are also valid input that should end to same result:  `+00:00` and `-00:00 `

Reply:
Error on my side - while datetime is case-insensitive as per spec, this is not the case for the time zone part.

Will add the suggested tests.

Reply:
One other funny test cases... add something on the dateline or even across - +/- 12h timezone exactly or  +14 (Kiribati), so even the date should change accordingly after the conversion.

Reply:
Done